src/main/java/me/itzg/tsdbcassandra/services/QueryService.java

```diff
@@ -2,68 +2,123 @@
 
 import static java.util.Objects.requireNonNull;
 
+import com.datastax.oss.driver.api.core.cql.Row;
+import java.time.Duration;
 import java.time.Instant;
-import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Map.Entry;
-import java.util.Objects;
+import me.itzg.tsdbcassandra.downsample.SingleValueSet;
+import me.itzg.tsdbcassandra.downsample.ValueSet;
+import me.itzg.tsdbcassandra.entities.Aggregator;
+import me.itzg.tsdbcassandra.model.MetricNameAndTags;
 import me.itzg.tsdbcassandra.model.QueryResult;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.cassandra.core.cql.ReactiveCqlTemplate;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
 @Service
 public class QueryService {
 
   private final ReactiveCqlTemplate cqlTemplate;
   private final MetadataService metadataService;
+  private final SeriesSetService seriesSetService;
 
   @Autowired
   public QueryService(ReactiveCqlTemplate cqlTemplate,
-                      MetadataService metadataService) {
+                      MetadataService metadataService,
+                      SeriesSetService seriesSetService) {
     this.cqlTemplate = cqlTemplate;
     this.metadataService = metadataService;
+    this.seriesSetService = seriesSetService;
   }
 
-  public Flux<QueryResult> query(String tenant, String metricName,
-                                 Map<String, String> queryTags,
-                                 Instant start, Instant end) {
+  public Flux<QueryResult> queryRaw(String tenant, String metricName,
+                                    Map<String, String> queryTags,
+                                    Instant start, Instant end) {
     return metadataService.locateSeriesSets(tenant, metricName, queryTags)
         .flatMapMany(Flux::fromIterable)
         .flatMap(seriesSet ->
```

Comment:
I'm just learning about reactor, so this may be a dumb question.  It looks like flatMap differs from map in that it splits out the mapped tasks into separate publishers.  Is that for parallelization?  Does the parallel scheduler get used automatically in that case?

Reply:
From what I understand it's really just the first part of your statement: splits out each item into a publisher each. Even though Reactor at first seems like it all about concurrency, it's apparently a secondary concern/benefit from the description at the start of https://projectreactor.io/docs/core/release/reference/#schedulers